src/sync/sync-utils.ts

```diff
@@ -39,6 +39,8 @@ export const findOrStartSync = async (
 		await RepoSyncState.deleteFromSubscription(subscription);
 	}
 
+	await resetRepoFailedCodes(subscription, syncType);
```

Comment:
I think we need to reset `failedCode` inside `resetTargetedTasks` function. That is place where we reset the status as well as cursor (for full sync only)